### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by tpreusse**
screen shot 2017-12-28 at 17 23 26
screen shot 2017-12-28 at 17 23 57

### CODE REVIEW

File: components/Profile/Contact.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+      (
+        !!value &&
+        !isWebUri(value) &&
+        t('profile/contact/publicUrl/error')
+      )
+    )
+  }
+]
+
+const AccessRoleDropdown = ({t, ...props}) => (
```
Comment (uxengine) [inline components/Profile/Contact.js]: 👍

File: components/Profile/Contact.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+          <IconLink
+            icon='mail'
+            href={`mailto:${user.email}`}
+          />
+        )}
+        {user.publicUrl && (
+          <IconLink
+            icon='link'
+            href={user.publicUrl}
+            target={'_blank'}
```
Comment (uxengine) [inline components/Profile/Contact.js]: target='_blank'

File: components/Profile/Edit.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+const styles = {
+  editLink: css({
+    display: 'block',
+    marginTop: 5
+  })
+}
+
+const EditLink = ({children, onClick, ...props}) =>
+  <a
+    href='#'
```
Comment (uxengine) [inline components/Profile/Edit.js]: Optional: Maybe use `href='#edit'` to give some basic a11y hint.

File: components/Profile/Edit.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+      e.preventDefault()
+      onClick(e)
+    }}
+    {...props}
+    {...linkRule}
+    {...styles.editLink}>
+    {children}
+  </a>
+
+const Edit = ({me, user, t, state, setState, startEditing, update}) => {
```
Comment (uxengine) [inline components/Profile/Edit.js]: No action required, but I think in the long run we'd benefit from documenting which of `state, setState, startEditing, update` is an object, a function or a bool (no wait, startEditing is a function, too). OK to leave that to more quiet times though ;)

File: components/Profile/Page.js [hunk 5]
```diff
@@ -0,0 +1,441 @@
+import React, { Component, Fragment } from 'react'
```
Comment (uxengine) [inline components/Profile/Page.js]: Just wondering if a `git mv` from index.js would have saved us some review diff here.

File: components/Profile/Portrait.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+      className={styles.dropzone.toString()}
+      style={{
+        cursor: isEditing
+          ? 'pointer' : 'auto'
+      }}
+      accept='image/jpeg, image/png, image/gif'
+      onDrop={(accepted, rejected) => {
+        if (accepted.length) {
+          const file = accepted[0]
+          if (file.size && file.size > 6200000) {
```
Comment (uxengine) [inline components/Profile/Portrait.js]: Maybe something like `5.91*1024*1024` would be more readable?

File: components/Profile/UsernameField.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+} from '@project-r/styleguide'
+
+const diacritics = [
+  {base: 'a', letters: ['ä', 'â', 'à']},
+  {base: 'c', letters: ['ç']},
+  {base: 'e', letters: ['é', 'ê', 'è', 'ë']},
+  {base: 'i', letters: ['î', 'ï']},
+  {base: 'o', letters: ['ö', 'ô']},
+  {base: 'u', letters: ['ü', 'ù', 'û']},
+  {base: 'ss', letters: ['ß']}
```
Comment (uxengine) [inline components/Profile/UsernameField.js]: Booo! ;)

File: lib/translations.json [hunk 8]
```diff
@@ -475,88 +435,208 @@
...
       "value": "Statement fehlt"
     },
     {
-      "key": "testimonial/pickImage",
+      "key": "profile/statement/tooLong",
+      "value": "Statement, maximal 140 Zeichen"
+    },
+    {
+      "key": "profile/preview",
+      "value": "Dies ist eine Vorschau von Ihrem Profil. Füllen Sie es jetzt aus und veröffentlichen Sie es."
```
Comment (uxengine) [inline lib/translations.json]: eine Vorschau Ihres Profils ... ?

File: lib/translations.json [hunk 9]
```diff
@@ -550,6 +550,14 @@
       "key": "profile/contact/phoneNumber/access/label",
       "value": "Telefonnummer freigeben für"
     },
+    {
+      "key": "profile/contact/isListed/label",
+      "value": "Portrait und Statement der Community-Seite anzeigen"
+    },
+    {
+      "key": "profile/contact/isAdminUnlisted/note",
+      "value": "Leider können wir Sie auf Community-Seite nicht anzeigen. Schreiben Sie an kontakt@republik.ch für weitere Informationen."
```
Comment (uxengine) [inline lib/translations.json]: ... auf **der** Community-Seite ...

File: components/Auth/withAuthorization.js [hunk 10]
```diff
@@ -77,9 +68,11 @@ const UnauthorizedPage = withT(({t, me, roles = []}) => (
   </Frame>
 ))
 
-export default roles => WrappedComponent => props => (
-  <EnsureAuthorization
-    roles={roles}
-    render={() => <WrappedComponent {...props} />}
-    unauthorized={({me}) => <UnauthorizedPage me={me} roles={roles} />} />
-)
+export const enforceAuthorization = roles => WrappedComponent => withAuthorization(roles)(({isAuthorized, me, ...props}) => {
```
[the remark this answers is not in the document]
Comment (tpreusse) [reply components/Auth/withAuthorization.js]: Currently only `enforceMembership`. But I'd like to leave it in for now—might be useful for an admin page.

[the remark this answers is not in the document]
Comment (uxengine) [reply components/Auth/withAuthorization.js]: SGTM

File: components/Profile/Edit.js (hunk 4 above)
Comment (tpreusse) [reply components/Profile/Edit.js]: added prop types

File: components/Profile/Page.js (hunk 5 above)
Comment (tpreusse) [reply components/Profile/Page.js]: not sure but would probably have marked almost all lines as changed anyhow

**Comment by tpreusse** [comment]
@uxengine thanks for the review of this rather huge PR!

File: components/Profile/Edit.js (hunk 4 above)
Comment (uxengine) [reply components/Profile/Edit.js]: 👍

**Pull request merged**